src/sinks/aws_cloudwatch_logs/config.rs

```diff
@@ -48,6 +49,44 @@ impl ClientBuilder for CloudwatchLogsClientBuilder {
     }
 }
 
+#[configurable_component]
+#[derive(Clone, Debug, Default)]
+/// Retention policy configuration for AWS CloudWatch Log Group
+pub struct Retention {
+    #[configurable(derived)]
+    #[serde(default)]
+    pub enabled: bool,
+
+    #[configurable(derived)]
+    #[serde(
+        default,
+        deserialize_with = "retention_days",
+        skip_serializing_if = "crate::serde::skip_serializing_if_default"
+    )]
+    pub days: i32,
```

Comment:
```suggestion
    pub days: u32,
```

Unfortunately, the docs aren't able to work with `i32`. `u32` should be fine here.